src/exchanger.js

```diff
@@ -3,90 +3,43 @@ var config = require("config");
-      callback(new Error("Randos not found"));
-      return;
-    }
-
-    logger.info("[exchanger.fetchAllRandosAsync]", "Exchanger.pair: Get ", randos.length, " randos");
-
-    async.eachLimit(randos, 30, function (rando, callback) {
-      logger.trace("[exchanger.fetchAllRandosAsync]", " Process rando: ", rando.randoId);
-      attachUserToRando(rando, function (err) {
-        logger.trace("[exchanger.fetchAllRandosAsync.attachUserToRando callback] ", "Done");
-        callback(err);
-      });
-    }, function (err) {
-      if (err) {
-        logger.warn("[exchanger.fetchAllRandosAsync]", "Each done with error: ", err);
-      } else {
-        logger.warn("[exchanger.fetchAllRandosAsync]", "Each done successfully");
-      }
-      callback(err, randos);
-    });
-  });
-}
-
-function attachUserToRando (rando, callback) {
-  db.user.getByEmail(rando.email, function (err, user) {
-    if (err) {
-      logger.debug("[exchanger.attachUserToRando] ", "Error on attachUserToRando: " + err);
-      callback(err);
-      return;
-    }
-
-    if (user) {
-      logger.trace("[exchanger.attachUserToRando] ", "Fetched user with email: ", user.email);
-      rando.user = user;
-      callback(err);
-    } else {
-      logger.debug("[exchanger.attachUserToRando] ", "User with email: " + rando.email + " not found");
-      callback(new Error("not found"));
-    }
-  });
-}
+global.users = {};
+global.randos = [];
 
-function exchangeRandos (randos, callback) {
+function exchangeRandos (callback) {
   logger.info("[exchanger.exchangeRandos]", "Trying exchange randos");
 
-  var choosers = findAllChoosers(randos);
-  printChooser(choosers);
+  var choosers = randoService.findAllChoosers(global.randos);
+  printService.printChooser(choosers);
 
   async.eachSeries(choosers, function (chooser, done) {
     //chooser is rando that will search other rando and put it to rando.user.in
     logger.debug("[exchanger.exchangeRandos]", "Chooser is:", chooser.randoId, "of [", chooser.email, "]");
 
     logger.trace("[exchanger.exchangeRandos]", "Start calculating by metrics");
-    metrics.calculate(chooser, randos);
+    metrics.calculate(chooser, global.randos);
 
     logger.trace("[exchanger.exchangeRandos]", "Calculation is done. Print metrics");
-    printMetrics(randos, chooser);
+    printService.printMetrics(global.randos, chooser);
 
     logger.trace("[exchanger.exchangeRandos]", "Trying to select best rando");
-    var bestRando = selectBestRando(randos);
+    var bestRando = randoService.selectBestRando(global.randos);
 
-    logger.debug("[exchanger.exchangeRandos]", "Best rando:", bestRando.randoId, "of", bestRando.user.email);
+    logger.debug("[exchanger.exchangeRandos]", "Best rando:", bestRando.randoId, "of", bestRando.email);
 
     if (bestRando.mark < 0) {
       logger.trace("[exchanger.exchangeRandos]", "Continue, because bestRando.mark < 0");
```

Comment:
We don't stop exchange we just skip current `chooser` and go to next. this line is a body of `eachSeries`:

```
20: async.eachSeries(choosers, function (chooser, done) {
```